scripts/box/migrate_to_external_account.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Script to migrate Box credentials from user settings object to external
+account objects.
+
+Changes:
+ - Create external account for authorized user settings
+ - Attach external account to user settings
+ - Attach external account to all node settings
+"""
+
+import sys
+import logging
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+
+from website.app import init_app
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+
+from website.addons.box.model import BoxUserSettings
+from website.addons.box.model import BoxNodeSettings
+from website.oauth.models import ExternalAccount
+
+logger = logging.getLogger(__name__)
+
+
+def do_migration(records):
+    database['boxnodesettings'].update({'user_settings': {'$type': 2}}, {'$rename': { 'user_settings': 'foreign_user_settings'}}, multi=True)
```

Comment:
Can you add a comment explaining why this field needs to be renamed? I know you mentioned a name conflict somewhere...but I can't seem to find where that is.